Slack-Apps/week1/homework/responses/1.3-setup-app-and-learn-events.md

```diff
@@ -0,0 +1,92 @@
+# Setting up a Slack App
+
+Now that you have done some research on Slack development, its time to get building! In this issue, you will learn how to set up a Slack app.
+
+## Installing a Text Editor
+
+Since your slack app will be runing locally, you will need a place to store and edit your code. You can create a folder anywhere within your file system for this project. If you do not have one already, it is highly reccomended that you download an IDE or text editor. This will give you a better interface for when you program, helping you to notice errors and run your code efficiently. Some good editors include [VSCode](https://code.visualstudio.com/download) or [Atom](https://atom.io/).
+
+## Creating a Slack App
+
+In order to create the app, you will need to [create a Slack account](https://slack.com/get-started#/create) and then make a [make a new app](https://api.slack.com/apps?new_app=1) on the developer platform. Once you have made the app, you should be able to click on it and navigate to a page similar to the one below:
+
+![Slack app home](./Gifs_Images/basic-info-page.png)
+
+
+
+## Installing Ngrok
+
+You might be wondering how your local code connects to the slack app. This process happens through a public facing url, or a server connection from your computer to the app. In order to make a public facing url, you first will need to [install ngrok](https://ngrok.com/) , which dedicates a server on your end to receiving requests from the slack app. 
+
+Open a the folder for your slack app in your editor, then open a terminal page. A terminal is a text interface for your computer's operating system. Type in ```pwd``` and press enter to make sure that you have navigated to the folder you created for the slack app. If you have not, refer to [this website](https://medium.com/@grace.m.nolan/terminal-for-beginners-e492ba10902a) on how to change folders. 
+
+Afterwards, refer to the [following tutorial](https://ngrok.com/) on installing ngrok. 
+
+
+
+## Connecting to the Slack App
+
+Now that you have installed and ran ngrok, you can connect your local code to your app. Copy the ```forwarding link``` of the running server, then navigate to the ```event subscriptions``` page in your app menu. Enable events, then copy and paste your link in the 'Request URL' field, followed by /slack/events. If your URL is not verified, try running ngrok again and using the new link.
```

Comment:
screenshots here might be helpful (and throughout the tutorial) in areas where you're telling students to find buttons